### PR TITLE
[CPU][ArmSME] Add `-arm-sme-vector-legalization` to ArmSME pipeline

### PR DESCRIPTION
**Pull request by MacDue**
This is needed to legalize vector types larger than a single SME tile, and for various small ArmSME-specific rewrites.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp
Comment (banach-space) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp]: This comment should be moved. Also, it would be good to add a note explaining what the legalization pass does :)

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

Please, lets have the comments updated before landing this :)

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp:704]: Do we run cleanup passes at the end of the if-statement?

Comment (MacDue) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp:704]: There's more cleanups that apply before things are lowered to ArmSME and SCF, so it's best to apply these right after the legalizer runs (which is still working on vector + arith ops).

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp:704]: I see, thanks

**Review by hanhanW — APPROVED** [review]

**Comment by MacDue** [comment]
Could someone land this for me (I don't have commit access). 

Thanks in advance :)

**Comment by Groverkss** [comment]
> Could someone land this for me (I don't have commit access).
> 
> Thanks in advance :)

Done! (Enabled auto merge)

**Pull request merged**
